docs/developer/architecture/coding-principles.md

```diff
@@ -27,26 +27,64 @@
 
 1. Avoid layers of indirection when they are not needed (e.g. "pass-through methods").
 
-### IV. Testing
+### IV. General Coding Style
+
+1. use `var` instead of explicit types (helps with clarity)
+2. avoid `final` in method args and local variables
+3. use `final` in field declarations
+4. avoid `static` fields except in constants or when absolutely necessary. (you should be able to provide a reason).
+5. use `interfaces` as containers for constants
+6. use "minimally required types" (or "smallest possible API"), e.g. use `ObjectMapper` instead of `TypeManager`
+   , or use a `String` instead of a more complex object containing the String, etc.
+7. use either `public` members, which are documented and tested, or `private` members.
+8. avoid package-private members, especially if only needed for testing
+9. avoid `protected` members unless they're intended to be overridden.
+10. use package-private classes if they're not needed outside the package, e.g. implementation classes
+11. avoid using `enum`s for anything other than named integer enumerations.
+12. avoid using static classes as much as possible. Exceptions to this are helper functions and test utils, etc.
+13. avoid any kind of variable prefixes, no underscores, no `m_` for members, etc.
+14. avoid unnecessary `this.` (exception: in builders)
+15. use static imports, as long as it is still clear what's being done. For example,
+    use `of(item1, item2).map(it -> it.someOperation)...` instead of `Stream.of(item1, item2)`, or `format("...",arg1)`
+    instead of `String.format("...", arg1)`. Helps with clarity and readability.
+16. avoid `Optional` as method return type or method argument. Use `null` in signatures.
+17. avoid cryptic variable names, especially in long methods. Instead, try to write them out, at least to a reasonable
```

Comment:
Do we need to say this, it's kind of like "don't write bad code"? This document should highlight our conventions, not how to write good code. Conventions come up when there is more than one coding idiom and we require one approach.

Reply:
developers sometimes have a tendency to abbreviate var names, e.g. `gat` instead of `generatedAccessToken`, I know I've been guilty of this, and with refactoring capabilities of modern IDEs it can happen all the more easily. That's why I put it in.